scripts/cre_sync/docs/implementation.md

```diff
@@ -0,0 +1,275 @@
+section: 1.1.1
+subsection: ""
+hyperlink: ""
+links:
+- type: 'SAM'
+  tags: []
+  document:
+    description: 'Some description'
+    doctype: CRE
+    id: 001-005-020
+    links: []
+    metadata: {}
+    name: CRE-NAME
+    tags: []
+    CREVersion: 2
+- type: 'SAM'
+  tags: []
+  document:
+    description: 'Another description'
+    doctype: CRE
+    id: 010-007 
+    links: []
+    metadata: {}
+    name: CRE007
+    tags: []
+    CREVersion: 2
+- type: 'SAM'
+  tags: []
+  document:
+    description: 'A standard Description'
+    doctype: Standard
+    id: ""
+    name: Top10-2017
+    section: 2
+    subsection: 3
+    links: []
+    metadata: {}
+    tags: []
+    version: 2
+    hyperlink: "https://blahblah"
+```
+
+This would tell the parser that ASVS 1.1.1 links to CRE001-005-020 and CRE010-007 and Top10-2017 item 2. All links are of the type 'SAM'
+
+The equivalent information CRE document, would be:
+
+```Yaml
+description: ''
+doctype: CRE
+id: 010-007
+name: 'Some Name'
+links:
+- type: 'SAM'
+  tags: []
+  document:
+    description: 'Some description'
+    doctype: CRE
+    id: 001-005-020
+    links: []
+    metadata: {}
+    name: CRE-NAME
+    tags: []
+    CREVersion: 2
+- type: 'SAM'
+  tags: []
+  document:
+    description: ''
+    doctype: Standard
+    id: ""
+    name: ASVS
+    section: 1.1.1
+    subsection: ""
+    hyperlink: ""
+- type: 'SAM'
+  tags: []
+  document:
+    description: 'A standard Description'
+    doctype: Standard
+    id: ""
+    name: Top10-2017
+    section: 2
+    subsection: 3
+    links: []
+    metadata: {}
+    tags: []
+    version: 2
+    hyperlink: "https://blahblah"
+```
+
+While this format is significantly longer than the one suggested in requirements.md , it allows for more complete mappings to be written. It also allows for the format to be preserved regardless of the user journey.
+A standards writer can submit the 'standard' example above and we can export/provide the 'CRE' format as they are different views of the same information.
+
+Mappings from a Spreadsheet (Sheet Parser)
+----
+
+Providing standards writers and community members the ability to submit a url linked to a Google spreadsheet and the application/parser import CRE mappings from there is a good idea, however care must be taken to ensure the spreadsheet is acutally parseable.
+In order for this to happen a robust template must be created and a parser from/to this template needs to be written.
+
+Since the design outlined in requirements.md is simple no more information is required at this stage.
+
+CRE WebApp & Query Engine
+---
+
+For CRE to be easily browsable a web application needs to be created. The application should implement the "refer" use case at a minimum.
+
+Backend
+---
+
+Internal Storage
+---
+
+A database needs to be created to hold all CRE information.
+The database management engine SQLITE3 has been chosen for simplicity and due to it's ability to be portable. This database has 4 tables
```

Comment:
here's my thoughts on this:
using mysql/postgres/neo4j  or anything else requires another daemon running somewhere server side, this makes the whole setup heavier.  The current setup can easily run in a couple Lambda functions without Aurora or other db support.

we don't have a significant amount of data yet so sqlite would be fast enough for our use case, 
it also allows for trivial development setup and an unnoficial "export" feature where we can just share the db with SKF or other integrators instead of files.

when DB performance starts being a problem we can switch to a real database